src/mono/wasm/runtime/binding_support.js

```diff
@@ -361,26 +361,7 @@ var BindingSupportLib = {
 			// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Typed_arrays
 			if (!!(js_obj.buffer instanceof ArrayBuffer && js_obj.BYTES_PER_ELEMENT)) 
 			{
-				var arrayType = 0;
-				if (js_obj instanceof Int8Array)
-					arrayType = 11;
-				if (js_obj instanceof Uint8Array)
-					arrayType = 12;
-				if (js_obj instanceof Uint8ClampedArray)
-					arrayType = 12;
-				if (js_obj instanceof Int16Array)
-					arrayType = 13;
-				if (js_obj instanceof Uint16Array)
-					arrayType = 14;
-				if (js_obj instanceof Int32Array)
-					arrayType = 15;
-				if (js_obj instanceof Uint32Array)
-					arrayType = 16;
-				if (js_obj instanceof Float32Array)
-					arrayType = 17;
-				if (js_obj instanceof Float64Array)
-					arrayType = 18;
-
+				var arrayType = js_obj[Symbol.for("wasm type")];
```

Comment:
this looks right, but are we sure it's always set?

Reply:
maybe throw an exception if it's missing, right now it'll just be passing undefined around

Reply:
It will be set in all the places it that were checked previously and the code undefined will marshal as 0 which was the fallback value before
https://github.com/dotnet/runtime/blob/ea57457e9f85829537147bbf9a472da20f2ba40c/src/mono/wasm/runtime/binding_support.js#L40

Reply:
I was explicitly trying not to change behavior here since I wasn't 100% sure if anything relied on it